### PR TITLE
[tx] Implement per request seeds

### PR DESCRIPTION
**Pull request by pbokc**
Make it possible for the user to specify a different seed for each request.

### CODE REVIEW

File: skyrl-tx/tests/utils/test_generator.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    input_ids = make_input_ids(batch_size=1, prompt_length=prompt_len)
+    attention_mask = make_attention_mask(batch_size=1, prompt_length=prompt_len)
+
+    sampling = make_sampling_param(max_tokens=4, temperature=1.0, seed=12345)
+
+    res1 = model.generate(input_ids, attention_mask, sampling_params=[sampling])
+    res2 = model.generate(input_ids, attention_mask, sampling_params=[sampling])
+
+    assert extract_generated_lists(res1) == extract_generated_lists(res2)
+    assert res1.logprobs == res2.logprobs
```
[the remark this answers is not in the document]
Comment (pbokc) [reply skyrl-tx/tests/utils/test_generator.py]: disagree - i want this to be a deterministic check

**Comment by pcmoritz** [comment]
Thanks a lot for the PR, it looks great! I'm going to make a few small modifications before merging it -- I agree the utils have become slightly cluttered and it will be good to reorganize some of the code. Note that circular dependencies are never a hard constraint (you can always break them by replacing `from ... import ...` with just `import ...` and then use the imported module in the code. So instead of solving for circular imports, it is more important to make sure there is a very clear mental model on where each part of the code lives and especially for new contributors to know where to put new functionality. I think we should try to move more stuff out of the utils and keep them really minimal (e.g. the checkpointing related stuff can be moved out, the generator can be moved out). Let's work on this in a separate PR though, for now I'm going to revert those changes so this PR can focus on one thing (it is always best to separate refactoring PRs from PRs that change the behavior of the code). I'll create an issue where we can discuss the code restructuring more :)
